### PR TITLE
Issue #434 - Forbid to have `+` signs before numbers

### LINKED ISSUE #434: Forbid to have `+` signs before numbers
# Rule request

## Thesis

You can write: `5.4` and `+5.4`.
There's no need to use the second version. Use the first one.

## Reasoning

This is done for consistency reasons.

### PR DESCRIPTION
**Pull request by MoskiMBA**
# I have made things!

## Checklist

- [x] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [x] I have created at least one test case for the changes I have made
- [ ] I have updated the documentation for the changes I have made
- [x] I have added my changes to the `CHANGELOG.md`

## Related issues
Closes #434
Refs #435 

I've named the visitor more generic, so it can be expaned in the future with issue #435 or related.
 Also, I'm not sure how to properly word the summary and reasoning in  `UselessOperatorsViolation`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ We used to have incremental versioning before `0.1.0`.
 - Forbids to have useless nodes
 - Forbids to have useless `raise` statements
 - Adds `params` and `parameters` to black-listed names
+- Forbids to use redundant `+` operator before numbers
```
Comment (sobolevn) [inline CHANGELOG.md]: Can you please create a new section `Version 0.8.0`? And put this line there.

File: tests/test_visitors/test_ast/test_builtins/test_useless_operators_before_numbers.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    visitor.run()
+
+    assert_errors(visitor, [UselessOperatorsViolation])
+
+
+@pytest.mark.parametrize('code', usages)
+@pytest.mark.parametrize('number', [
+   '5',
+   '-0.5',
+   '+-0x20',
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_builtins/test_useless_operators_before_numbers.py]: Why this number is correct?

File: tests/test_visitors/test_ast/test_builtins/test_useless_operators_before_numbers.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+   '++-0o12',
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_builtins/test_useless_operators_before_numbers.py]: And this one.

File: tests/test_visitors/test_ast/test_builtins/test_useless_operators_before_numbers.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+   '+~0.5',
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_builtins/test_useless_operators_before_numbers.py]: And this one.

File: tests/test_visitors/test_ast/test_builtins/test_useless_operators_before_numbers.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+function_definition = """
+def function_name(param1, param2={0}):
+    return param1 / param2
+"""
+
+list_definition = '[{0}]'
+dict_definition_key = '{{{0}: "value"}}'
+dict_definition_value = '{{"first": {0}}}'
+set_definition = '{{"first", {0}, "other"}}'
+tuple_definition = '({0}, )'
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_builtins/test_useless_operators_before_numbers.py]: We also need to test `x + y` and `x += 1` cases. They should be allowed.

File: wemake_python_styleguide/violations/consistency.py [hunk 6]
```diff
@@ -1141,3 +1143,34 @@ class UselessExceptCaseViolation(ASTViolation):
 
     error_template = 'Found useless `except` case'
     code = 329
+
+
+class UselessOperatorsViolation(ASTViolation):
+    """
+    Forbids the use of unnecessary operators in your code.
+
+    You can write: ``5.4`` and ``+5.4``. There's no need to use the second
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/consistency.py]: Can you please rephrase it to be more readable? Since we only support `+` for now, you can only mention `+` operators.

File: wemake_python_styleguide/types.py [hunk 7]
```diff
@@ -60,6 +60,14 @@
 #: Tuple of AST node types for declarative syntax.
 AnyNodes = Tuple[Type[ast.AST], ...]
 
+#: In cases we need to work with all unary operators
+AnyUnaryOp = Union[
```
[the remark this answers is not in the document]
Comment (sobolevn) [reply wemake_python_styleguide/types.py]: When doing `isinstance(parent.op, operator)` where `operator` is `Union` - `mypy` will complain.

`isinstance` waits for `Tuple[...]`.
So, my guess would be:

1. Leave `AnyUnaryOp` as it is
2. Refactor `if isinstance(parent.op, operator)` to be `if isinstance(parent.op, (operator,))`

[the remark this answers is not in the document]
Comment (MoskiMBA) [reply wemake_python_styleguide/types.py]: I've refactored this line to `if isinstance(parent.op, (operator,))`

**Comment by sobolevn** [comment]
@MoskiMBA now `0.7.1` is released, and I will merge your changes soon! Thanks for building it!

**Comment by sobolevn** [comment]
@MoskiMBA thanks for working on this issue! It is very valuable and useful.
Merging it into `0.8.0`!

🎉
